seurat-find-transfer-anchor.R

```diff
@@ -0,0 +1,220 @@
+    type = 'character',
+    help = "assay to use from query"
+  ),
+  make_option(
+    c("--reduction"),
+    action = "store",
+    default = "pcaproject",
+    type = 'character',
+    help = "Dimensional reduction to perform when finding anchors.  "
+  ),
+  make_option(
+    c("--project-query"),
+    action = "store",
+    default = FALSE,
+    type = 'logical',
+    help = "Project the PCA from the query dataset onto the reference. Use only in rare cases "
+  ),
+  make_option(
+    c("--features"),
+    action = "store",
+    default = NULL,
+    type = 'character',
+    help = "Features to use for dimensional reductionFeatures to use for dimensional reduction"
+  ),
+  make_option(
+    c("--npcs"),
+    action = "store",
+    default = 30,
+    type = 'integer',
+    help = "Number of PCs to compute on reference. If null, then use an existing PCA structure in the reference object"
+  ),
+  make_option(
+    c("--dims"),
+    action = "store",
+    default = 0:30,
+    type = 'integer', #note sure
+    help = "dimension to go throw"
+  ),
+  make_option(
+    c("--k-anchor"),
+    action = "store",
+    default = 5,
+    type = 'integer',
+    help = "How many neighbors (k) to use when picking anchors "
+  ),
+  make_option(
+    c("--k-filter"),
+    
+    action = "store",
+    default = 200,
+    type = 'integer',
+    help = "How many neighbors (k) to use when filtering anchors "
+  ),
+  make_option(
+    c("--k-score"),
+    action = "store",
+    default = 30,
+    type = 'integer',
+    help ="How many neighbors (k) to use when scoring anchors"
+  ),
+  make_option(
+    c("--max-features"),
+    action = "store",
+    default = 200,
+    type = 'integer',
+    help ="The maximum number of features to use when specifying the neighborhood search space in the anchor filtering"
+  ),
+  make_option(
+    c("--nn-method"),
+    action = "store",
+    default = "rann",
+    type = 'character',
+    help ="Method for nearest neighbor finding. Options include: rann, annoy"
+  ),
+  make_option(
+    c("--eps"),
+    action = "store",
+    default = 0,
+    type = 'integer',
+    help ="Error bound on the neighbor finding algorithm (from RANN)"
+  ),
+  make_option(
+    c("--approx-pca"),
+    action = "store_true",
+    default = TRUE,
+    help ="Use truncated singular value decomposition to approximate PCA"
+  ),
+  make_option(
+    c("--verbose"),
+    action = "store_true",
+    default = TRUE,
+    help ="Print progress bars and output"
+  )
+  
+)
+
+#minimum arguments to work
+opt <- wsc_parse_args(option_list, mandatory = c('query_file', 'reference_file','output_file'))
+
+# Check parameter values
+
+if ( ! file.exists(opt$reference_file)){
+  stop((paste('File', opt$reference_file, 'does not exist')))
+}
+
+
+#load seurat and packages needed to read input
+suppressPackageStartupMessages(require(Seurat))
+if(opt$query_format == "loom" | opt$output_format == "loom") {
+  suppressPackageStartupMessages(require(loomR))
```

Comment:
Is the result of this operation an actual seurat/clustering object, or something else like a file encoding mappings between query and reference? If it is the latter you might not need this part.

Reply:
The output is an "AnchorSet class" with list and matrix in it so I think it is a clustering object 

![image](https://user-images.githubusercontent.com/43669368/76779451-47ca6c80-67a3-11ea-9928-d93a5282d19d.png)